### PR TITLE
Issue/10516 campaign preview screen UI

### LINKED ISSUE #10538: Preview campaign content UI
Part of: #10516

### PR DESCRIPTION
**Pull request by JorgeMucientes**
Part of: https://github.com/woocommerce/woocommerce-android/issues/10516

Closes: #10538

### Description

Adds the UI for Blaze campaign preview UI. No click events are handled. The data displayed is hardcoded. No real data is being retrieved or used. 

Note there's a known bug I'm still looking into. More on it at the end.

### Testing instructions

1. Log into a site eligible for Blaze
2. Click on Create Campaign. 
3. Review the UI for the campaign preview base on this Figma: mBhyoV7OPhFnOugxLy6SMS-fi-2506%3A53905

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/2663464/b912b056-af30-4664-a387-e540528cbdd6

### Known issue

When navigating to campaign preview screen from product selector there's an issue where the toolbar from `My Store tab` is shown on top of the toolbar from the campaign preview screen. See screen recording: 

https://github.com/woocommerce/woocommerce-android/assets/2663464/11f70a22-a855-49a0-a12d-ba1bfb3b1d46

The reason for this is how we handle navigation after selecting a product from the product selector. What happens internally in the above video is: 
1. When a product is selected from the product selector, the product selector is dismissed and the result is returned to the previous fragment, in this case `MyStoreFragment`
2. `MyStoreFragment` handles the result through `BlazeCampaignCreationDispatcher.attachFragment()` that triggers blaze campaign creation screen passing the selected productId. 

The problem is that between steps 1 and 2, `MyStoreFragment` is reloaded and `onViewCreated` called again. This triggers the animation to expand its toolbar. This animation is enqueued and  handled by `MainActivity.onFragmentViewCreated`. By the time the expand animation is run from MainActivity (the host activity for `MyStoreFragment` as well as `BlazeCampaignCreationPreviewFragment`), `MyStoreFragment` is not visible anymore and `BlazeCampaignCreationPreviewFragment` is, resulting in displaying the expanded toolbar for the campaign creation fragment.

Will try to figure something out to fix that in a different PR

- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>BlazeCampaignCreationPreviewViewModel</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>BlazeCampaignCreationPreviewViewModel</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>BlazeCampaignCreationPreviewViewModel</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>BlazeCampaignCreationPreviewViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>BlazeCampaignCreationPreviewViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/188">17.0</a>. The due date for this milestone has already passed.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/MyStoreBlazeViewModel.kt [hunk 1]
```diff
@@ -41,7 +40,6 @@ class MyStoreBlazeViewModel @Inject constructor(
     private val analyticsTrackerWrapper: AnalyticsTrackerWrapper,
     private val prefsWrapper: AppPrefsWrapper
 ) : ScopedViewModel(savedStateHandle) {
-    @OptIn(ExperimentalCoroutinesApi::class)
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/MyStoreBlazeViewModel.kt:44]: Unrelated to the PR. This was simply unneeded here

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/preview/PreviewAnnotations.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+@Preview(name = "Portrait Mode", showBackground = true, device = Devices.PIXEL_4)
+annotation class OrientationPreviews
+
+@Preview(name = "Default Font Size", fontScale = 1f)
+@Preview(name = "Large Font Size", fontScale = 1.5f)
+annotation class FontScalePreviews
+
+@Preview(name = "Left-To-Right", locale = "en")
+@Preview(name = "Right-To-Left", locale = "ar")
+annotation class LayoutDirectionPreviews
```
Comment (JorgeMucientes) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/preview/PreviewAnnotations.kt:22]: Some of these annotations are not used yet, but I think they will be handy in the future. They provide much cleaner annotations for compose previews. See the example in BlazeCampaignPreviewScreen: 

```kotlin
@LightDarkThemePreviews
@Composable
fun CampaignScreenPreview() {
```

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>BlazeCampaignCreationPreviewViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/188">17.0</a>. The due date for this milestone has already passed.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>BlazeCampaignCreationPreviewViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/188">17.0</a>. The due date for this milestone has already passed.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewScreen.kt [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+            Text(
+                text = title,
+                style = MaterialTheme.typography.subtitle1,
+                color = colorResource(id = R.color.color_on_surface_high)
+            )
+            Text(
+                text = content,
+                style = MaterialTheme.typography.body2,
+                color = colorResource(id = R.color.color_on_surface_medium)
+            )
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewScreen.kt]: A bit nitpicky but for "Ad destination" it seems like the design wants this to just be one line, with "..." at the text cut-off point.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewScreen.kt [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+fun CampaignPreviewContent(
+    state: CampaignPreviewContent,
+    modifier: Modifier = Modifier,
+) {
+    Column(
+        modifier = modifier
+            .verticalScroll(rememberScrollState())
+            .fillMaxSize()
+    ) {
+        CampaignHeader(
```
Comment (hafizrahman) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewScreen.kt:95]: This area could use a bit more styling in dark mode, currently:

Screenshot 2024-01-12 at 22 08 06

Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewScreen.kt:95]: I ran out of time to do more review today but I'll be back tomorrow with more.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/preview/PreviewAnnotations.kt (hunk 2 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/compose/preview/PreviewAnnotations.kt:22]: Very handy. The LayoutDirectionPreviews will be useful for checking e.g: the right arrow on each row that is used in the design here as well.

**Review by hafizrahman — CHANGES_REQUESTED** [review]
Some things I found that might need updating:

1. Dark mode needed for the Card,
2. Ad destination button design tweak

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewScreen.kt (hunk 4 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewScreen.kt:95]: Hey @hafizrahman thanks for the review! Great input. I'll start addressing your feedback now. I do have one question, though. Can you elaborate on this? 

> This area could use a bit more styling in dark mode, currently

You can find the dark mode version of this header in Figma: mBhyoV7OPhFnOugxLy6SMS-fi-2845_46941

Is there anything missing?

Screenshot 2024-01-15 at 09 55 19

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>BlazeCampaignCreationPreviewViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/188">17.0</a>. The due date for this milestone has already passed.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewScreen.kt (hunk 3 above)
Comment (JorgeMucientes) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewScreen.kt]: Good catch! I had to do some extra refactoring to support this and simplify the UI logic. But now the destination URL is enforced to a single-line value!

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Class <code>BlazeCampaignCreationPreviewViewModel</code> is missing tests, but <code>unit-tests-exemption</code> label was set to ignore this.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewScreen.kt (hunk 4 above)
Comment (hafizrahman) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/blaze/creation/preview/BlazeCampaignCreationPreviewScreen.kt:95]: Huh, I'm surprised this has pure white bg in dark mode even in the design, I didn't realize. Ignore my feedback then.

**Review by hafizrahman — APPROVED** [review]
Thanks for the changes (and for correcting me on the white preview box). This is good to go now.

**Pull request merged**